### PR TITLE
New Kubernetes Kueue Operators

### PR DESCRIPTION
**Pull request by moiseenkov**
- implement new operators: `KubernetesInstallKueueOperator`, `KubernetesStartKueueJobOperator` in `cncf-kubernetes` provider
- refactor GKE operators in `google` provider in order to reduce complexity and code duplication

### CODE REVIEW

File: providers/src/airflow/providers/cncf/kubernetes/operators/kueue.py [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+                " in a Kubernetes cluster, set the `suspend` parameter to True.",
+            )
+        elif self.suspend is None:
+            warnings.warn(
+                f"You have not set parameter `suspend` in class {self.__class__.__name__}. "
+                "For running a Job in Kueue the `suspend` parameter should set to True.",
+                UserWarning,
+                stacklevel=2,
+            )
+            self.suspend = True
```
Comment (eladkal) [inline providers/src/airflow/providers/cncf/kubernetes/operators/kueue.py]: This operator is new code, so why do we need to warn users about wrong usage? We can simply "force" the right use from the start

File: providers/tests/deprecations_ignore.yml [hunk 2]
```diff
@@ -94,7 +94,6 @@
 - providers/tests/google/cloud/operators/test_dataproc.py::test_scale_cluster_operator_extra_links
 - providers/tests/google/cloud/operators/test_dataproc.py::test_submit_spark_job_operator_extra_links
 - providers/tests/google/cloud/operators/test_gcs.py::TestGoogleCloudStorageListOperator::test_execute__delimiter
-- providers/tests/google/cloud/operators/test_kubernetes_engine.py::TestGoogleCloudPlatformContainerOperator::test_create_execute_error_body
```
Comment (eladkal) [inline providers/tests/deprecations_ignore.yml:97]: How is this change related to the PR?

File: providers/src/airflow/providers/cncf/kubernetes/operators/kueue.py [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+
+from kubernetes.utils import FailToCreateError
+
+from airflow.exceptions import AirflowException
+from airflow.models import BaseOperator
+from airflow.providers.cncf.kubernetes.hooks.kubernetes import KubernetesHook
+from airflow.providers.cncf.kubernetes.operators.job import KubernetesJobOperator
+
+
+class KubernetesInstallKueueOperator(BaseOperator):
```
Comment (eladkal) [inline providers/src/airflow/providers/cncf/kubernetes/operators/kueue.py:33]: Can you please add docs for all the new features in this PR?

File: providers/src/airflow/providers/cncf/kubernetes/operators/kueue.py (hunk 1 above)
Comment (moiseenkov) [reply providers/src/airflow/providers/cncf/kubernetes/operators/kueue.py]: Good point! I replaced the warning by informing here, so users wouldn't be surprised that the `suspend` parameter was forcibly changed.

File: providers/src/airflow/providers/cncf/kubernetes/operators/kueue.py (hunk 3 above)
Comment (moiseenkov) [reply providers/src/airflow/providers/cncf/kubernetes/operators/kueue.py:33]: Done.

File: providers/tests/deprecations_ignore.yml (hunk 2 above)
Comment (moiseenkov) [reply providers/tests/deprecations_ignore.yml:97]: This unit test was replaced by the new test during GKE operators refactoring. And because it's no longer present, we don't have to ignore it anymore.

**Comment by moiseenkov** [comment]
@eladkal , @potiuk,
 
This PR contains changes for both Google and cncf-kubernetes providers, and the Google provider depends on changes in cncf-kuberneets. For this reason I had to bump cncf-kubernetes version in Google provider's [dependencies](https://github.com/apache/airflow/pull/44568/files#diff-bfccba1ff8b4b3e462131ef3d0869d7760d64a2f3b6f1a15b89157a4c7d68404L191) and it now breaks CI because it can't install cncf-kubernetes provider that wasn't released yet.
Therefore, should I split this PR, so we could release these changes gradually (cncf first, and google later) or we are OK with merging it all together now.

**Comment by potiuk** [comment]
> @eladkal , @potiuk,
> 
> This PR contains changes for both Google and cncf-kubernetes providers, and the Google provider depends on changes in cncf-kuberneets. For this reason I had to bump cncf-kubernetes version in Google provider's [dependencies](https://github.com/apache/airflow/pull/44568/files#diff-bfccba1ff8b4b3e462131ef3d0869d7760d64a2f3b6f1a15b89157a4c7d68404L191) and it now breaks CI because it can't install cncf-kubernetes provider that wasn't released yet. Therefore, should I split this PR, so we could release these changes gradually (cncf first, and google later) or we are OK with merging it all together now.

This is one option yes. There is another option. You can add the `cncf.kubernetes` provider to the list of "chicken-egg" providers in `src/airflow_breeze/global_constants.py` ->  similarly to what I've done in https://github.com/apache/airflow/pull/44714 (about to be merged).

**Comment by potiuk** [comment]
Also updated documentation about it here https://github.com/apache/airflow/pull/44720

**Comment by eladkal** [comment]
> > @eladkal , @potiuk,
> > This PR contains changes for both Google and cncf-kubernetes providers, and the Google provider depends on changes in cncf-kuberneets. For this reason I had to bump cncf-kubernetes version in Google provider's [dependencies](https://github.com/apache/airflow/pull/44568/files#diff-bfccba1ff8b4b3e462131ef3d0869d7760d64a2f3b6f1a15b89157a4c7d68404L191) and it now breaks CI because it can't install cncf-kubernetes provider that wasn't released yet. Therefore, should I split this PR, so we could release these changes gradually (cncf first, and google later) or we are OK with merging it all together now.
> 
> This is one option yes. There is another option. You can add the `cncf.kubernetes` provider to the list of "chicken-egg" providers in `src/airflow_breeze/global_constants.py` -> similarly to what I've done in #44714 (about to be merged).


Why not just using AirflowOptionalProviderFeatureException?

**Comment by moiseenkov** [comment]
> This is one option yes. There is another option. You can add the `cncf.kubernetes` provider to the list of "chicken-egg" providers in `src/airflow_breeze/global_constants.py` -> similarly to what I've done in #44714 (about to be merged).

Thank you for the suggestion! I followed this idea.

> Why not just using AirflowOptionalProviderFeatureException?

I've also considered this, but unfortunately provided changes aren't optional. For example the `GKEStartKueueInsideClusterOperator` now inherits from `KubernetesInstallKueueOperator` which was introduced in this PR. At the same time the `GKEStartKueueInsideClusterOperator` is a bit older and will be broken if the Kubernetes provider isn't upgraded. Thus I had to bump the cncf-kubernetes provider version in google provider's dependencies.

File: providers/src/airflow/providers/google/provider.yaml [hunk 4]
```diff
@@ -188,7 +188,7 @@ additional-extras:
       - apache-beam[gcp]
   - name: cncf.kubernetes
     dependencies:
-      - apache-airflow-providers-cncf-kubernetes>=7.2.0
+      - apache-airflow-providers-cncf-kubernetes>10.0.1
```
Comment (potiuk) [inline providers/src/airflow/providers/google/provider.yaml]: Actually this should be 10.1.0 (and cncf.kuberbetes should also be bumped to 10.1.0 in their provider.yaml). We are effectively working on a "future" version of the cncf.kubernetes provider, which adds new features, it's version will be 10.1.0 - and we should already do it now as part of this change. This the only exception when provider version should be modified by someone else than release manager.

Comment (moiseenkov) [reply providers/src/airflow/providers/google/provider.yaml]: Awesome! Bumped the version

**Comment by potiuk** [comment]
Tests failing - breeze (easier to fix) and compatibility tests with airflow 2.8 - 2.9 (a bit more difficult). The way how to run compatibilty test is described in detail in https://github.com/apache/airflow/blob/main/contributing-docs/testing/unit_tests.rst#compatibility-provider-unit-tests-against-older-airflow-releases - with some examples how to deal with the tests.

**Comment by moiseenkov** [comment]
Hi @potiuk ,
The CI is finally green. Thank you very much for your guidance!
The PR is ready for review and merging hopefully.

**Review by potiuk — APPROVED** [review]
